content/en/blog/_posts/2023-04-18-node-expand-secret-moves-beta

```diff
@@ -0,0 +1,128 @@
+Request from the Container Orchestrator like Kubernetes in absence of this feature, thus CSI driver fails to perform the operation on such clusters. This made a blocker for many storage drivers to expand the volumes in Node level which is a critical feature for workload units to get more space in the persistent volume when required or on demand!
+
+It is worth mentioning that, it is not only the node level expansion operation which was problematic but we have seen use cases like below:
+
+At times, the CSI driver needs to check the actual size of the backend block storage (or image) before proceeding with a node-level filesystem expand operation. This avoids false positive returns from the backend storage cluster during file system expansion.
+When a PersistentVolume represents encrypted block storage (for example using LUKS) you need to provide a passphrase in order to expand the device, and also to make it possible to grow the filesystem on that device.
+
+What’s new in Beta?
+
+With the promotion of Volume Snapshot to beta, the feature is now enabled by default on standard Kubernetes deployments instead of being opt-in, that  means, no need to enable the  feature gate in the control plane components which was required on this feature’s alpha status. As part of promoting this feature to Beta staus, the support has also added in the CSI sidecar controller external-provisioner . The external provisioner sidecar controller has to be on v3.3.0 or above to take advantage of this feature. 
+
+How do I use CSINodeExpandSecret feature?
+
+Assuming all the required components (including CSI driver) are already deployed and running on your cluster, you can try NodeExpansion operation on a CSI volume if the CSI driver support the same. 
+
+The credentials for CSI NodeExpand Secret operation can be provided as Kubernetes Secret Object and specifying via StorageClass that use that Secret.
+
+Here is an example:
+
+manifest for a Secret that holds credentials:
+
+
+apiVersion: v1
+kind: Secret
+metadata:
+ name: test-secret
+ namespace: default
+data:
+stringData:
+ username: admin
+ password: t0p-Secret
+
+Here's an example manifest for a StorageClass that refers to those credentials:
+
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+ name: csi-blockstorage-sc
+parameters:
+ csi.storage.k8s.io/node-expand-secret-name: test-secret  # the name of the Secret
+ csi.storage.k8s.io/node-expand-secret-namespace: default # the namespace that the Secret is in
+provisioner: blockstorage.cloudprovider.example
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+allowVolumeExpansion: true
```

Comment:
Wrap this in <tt>\`\`\`yaml</tt> / <tt>\`\`\`</tt>, and make the first line be `---`.